indigo_app/forms.py

```diff
@@ -500,27 +520,66 @@ def filter_queryset(self, queryset):
-                .filter(commencements__count__gt=1)
-        elif self.cleaned_data.get('commencement') == 'range':
-            if self.cleaned_data.get('commencement_date_start') and self.cleaned_data.get('commencement_date_end'):
-                start_date = self.cleaned_data['commencement_date_start']
-                end_date = self.cleaned_data['commencement_date_end']
-                queryset = queryset.filter(commencements__date__range=[start_date, end_date]).order_by('-commencements__date')
-
+        # filter by commencement status
+        if exclude != "commencement":
+            commencement_filter = self.cleaned_data.get('commencement', [])
+            commencement_qs = Q()
+            if 'yes' in commencement_filter:
+                commencement_qs |= Q(commenced=True)
+            if 'no' in commencement_filter:
+                commencement_qs |= Q(commenced=False)
+            if 'date_unknown' in commencement_filter:
+                commencement_qs |= Q(commencements__date__isnull=True, commenced=True)
+
+            queryset = queryset.filter(commencement_qs)
+
+        if self.cleaned_data.get('commencement_date_start') and self.cleaned_data.get('commencement_date_end'):
+            start_date = self.cleaned_data['commencement_date_start']
+            end_date = self.cleaned_data['commencement_date_end']
+            queryset = queryset.filter(commencements__date__range=[start_date, end_date]).order_by('-commencements__date')
+
+        # filter by consolidation
+        if exclude != "consolidation":
+            consolidation_filter = self.cleaned_data.get('consolidation', [])
+            consolidation_qs = Q()
+            if 'has_consolidation' in consolidation_filter:
+                consolidation_qs |= Q(arbitrary_expression_dates__date__isnull=False)
+            if 'no_consolidation' in consolidation_filter:
+                consolidation_qs |= Q(arbitrary_expression_dates__date__isnull=True)
+
+            queryset = queryset.filter(consolidation_qs)
+
+        # filter by points in time
+        if exclude != "documents":
+            documents_filter = self.cleaned_data.get('documents', [])
+            documents_qs = Q()
+            if 'one' in documents_filter:
+                one_document_ids = queryset.annotate(Count('document')).filter(document__count=1).values_list('pk', flat=True)
+                documents_qs |= Q(id__in=one_document_ids)
+            if 'multiple' in documents_filter:
+                multiple_document_ids = queryset.annotate(Count('document')).filter(document__count__gt=1).values_list('pk', flat=True)
+                documents_qs |= Q(id__in=multiple_document_ids)
+            if 'none' in documents_filter:
+                documents_qs |= Q(document__isnull=True)
+            if 'published' in documents_filter:
+                documents_qs |= Q(document__draft=False)
+            if 'draft' in documents_filter:
+                documents_qs |= Q(document__draft=True)
```

Comment:
published and draft are done separately under status, so these will never be in?